### DISCUSSION TITLE
What is "index.runtime.*.js"?

### DISCUSSION THREAD

**Discussion by jaredkrinke** [parcel-bundler/parcel#9490] [Q&A]
I have a trivial HTML file which only includes a single script (although that script transitively includes quite a bit!), and I don't think I used to see this, but it's now generating two scripts: "index.\*.js" (expected) and "index.runtime.\*.js" (which I'm curious about).

My trivial HTML:

```html
<!DOCTYPE html>
<html>
    <head>
        <meta charset="utf-8">
        <title>SIC-1</title>
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
        <link rel="stylesheet" href="css/main.css" />
        <link rel="shortcut icon" href="favicon.ico" />
    </head>
    <body>
        <div id="root"></div>
        <script type="module" src="ts/main.tsx"></script>
    </body>
</html>
```

I searched the web, issues, and discussions and couldn't find an explanation. I'm probably wrong, but I don't *think* I used to see this file.

For what it's worth, I wasn't including the file in my Electron app (because I didn't know about it) and nothing obvious was breaking. It's a pretty short file, but it's minified so not obvious what it's doing. The script is included in my HTML `<head>`:

```html
<!DOCTYPE html><html><head><script type="module" src="index.runtime.c5bb2c65.js"></script>
```

**Comment by jaredkrinke** [comment]
Poking around the source map and Parcel source code, I think this is the "prelude", but I'm not yet sure what it's for and whether or not it's ok that my release doesn't contain it (other than logging an error, of course).

Edit: also confirmed that this was not being included before.

**Comment by jaredkrinke** [comment]
Related question: is there a way to avoid/disable whatever generates this file?

**Comment by jaredkrinke** [comment]
I think I found the relevant code: https://github.com/parcel-bundler/parcel/blob/874dddf3eeeb1a57cb53f19bfdc171401f53f579/packages/core/core/src/applyRuntimes.js#L43C1-L61C2

I suspect it is for "runtime" plugins, but it looks like runtime plugins are in the default configuration, so I'm not sure I can avoid having this file generated.

Tangentially related to my question: the reason I was trying to find information about this file is that it wasn't having its name adjusted by a namer plugin I'm using, and I think this comment indicates this is known behavior:

```javascript
  // We don't run custom namers on runtime bundles as the runtime assumes that they are
  // located at the same nesting level as their owning bundle. Custom naming could
  // be added in future as long as the custom name is validated.
```

Edit: this comment has some information as well: https://github.com/parcel-bundler/parcel/issues/9309#issuecomment-1898364451

**Comment by stevexbritton** [comment]
I too would love to know how to prevent this as it is preventing me from upgrading Parcel beyond version 2.8.3. Parcel kindly generates a dynamic blob when it needs to import a worker script from another origin, which worked great in version 2.8.3, but now it doesn't work because it doesn't know handle this intermediate runtime script its generated during the build.

**Reply by jaredkrinke** [reply]
In my case, I was able to avoid this file using the workaround noted in this comment:

https://github.com/parcel-bundler/parcel/issues/9309#issuecomment-1898364451
